src/protocol.rs

```diff
@@ -347,23 +419,22 @@ impl<'a> SpanContents<'a> for MietteSpanContents<'a> {
     }
 }
 
-/**
-Span within a [`SourceCode`] with an associated message.
-*/
+/// Span within a [`SourceCode`]
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct SourceSpan {
     /// The start of the span.
     offset: SourceOffset,
-    /// The total length of the span. Think of this as an offset from `start`.
-    length: SourceOffset,
+    /// The total length of the span
+    length: usize,
 }
 
 impl SourceSpan {
     /// Create a new [`SourceSpan`].
     pub fn new(start: SourceOffset, length: SourceOffset) -> Self {
```

Comment:
It should be:
```suggestion
    pub fn new(start: SourceOffset, length: usize) -> Self {
```
But will be breaking change. 

`SourceOffset` may not represent length in no way. This looks more like a range (e.g `3..3`)

Reply:
Can you make a pr for this once this lands? I'll mark it as breaking but that way we won't forget to do it. 

There's a good chunk of desirable breaking changes in the pipeline right now and I think I'm gonna find some time to roll up 6.0 at some point.

Reply:
Ok